stdlib/database/sql/src/main/ballerina/src/sql/error.bal

```diff
@@ -14,6 +14,23 @@
 // specific language governing permissions and limitations
 // under the License.
 
+public const BATCH_EXECUTE_ERROR_REASON = "{ballerina/sql}BatchExecuteError";
+
+# Represents the properties belonging to a `BatchExecuteError`.
+#
+# + message - Error message
+# + errorCode - SQL error code
+# + sqlState - SQL state
+# + cause - Cause of the error
+# + executionResults - Result of execution of commands.
+public type BatchExecuteErrorData record {|
+    string message?;
+    int errorCode;
+    string sqlState;
+    error cause?;
+    ExecutionResult[] executionResults;
+|};
+
 public const DATABASE_ERROR_REASON = "{ballerina/sql}DatabaseError";
```

Comment:
Missing docs.

Reply:
These const are used for internal error creation and will be removed after the error syntax migration is complete